sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  Controller,
+  Param,
+  UnprocessableEntityException,
+  Get,
+  ForbiddenException,
+} from "@nestjs/common";
+import { HasLocationAccess, AllowAuthorizedParty } from "../../auth/decorators";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import {
+  ApplicationService,
+  InstitutionService,
+  InstitutionLocationService,
+} from "../../services";
+import { Application } from "../../database/entities";
+import { UserToken } from "../../auth/decorators/userToken.decorator";
+import { IInstitutionUserToken } from "../../auth/userToken.interface";
+import { COESummaryDTO } from "../application/models/application.model";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("institution/location")
+export class ConfirmationOfEnrollmentController {
+  constructor(
+    private readonly applicationService: ApplicationService,
+    private readonly institutionService: InstitutionService,
+    private readonly locationService: InstitutionLocationService,
+  ) {}
+
+  /**
+   * Get all application of a location in a institution
+   * with Confirmation Of Enrollment(COE) status completed and required
+   * @param locationId location that is completing the COE.
+   * @returns student application list of an institution location
+   */
+  @AllowAuthorizedParty(AuthorizedParties.institution)
+  @HasLocationAccess("locationId")
+  @Get(":locationId/confirmation-of-enrollment")
+  async getCOESummary(
+    @Param("locationId") locationId: number,
+    @UserToken() userToken: IInstitutionUserToken,
+  ): Promise<COESummaryDTO[]> {
+    const institutionDetails =
+      await this.institutionService.getInstituteByUserName(userToken.userName);
+    const requestedLoc = await this.locationService.getInstitutionLocationById(
+      locationId,
+    );
+    if (!institutionDetails) {
+      throw new UnprocessableEntityException(
+        "Not able to find a institution associated with the current user name.",
+      );
+    }
+    if (institutionDetails.id !== requestedLoc.institution.id) {
+      throw new ForbiddenException();
+    }
+    const applications = await this.applicationService.getCOEApplications(
+      locationId,
+    );
+    return applications.map((eachApplication: Application) => {
+      return {
```

Comment:
IMO, considering adding the null checks where they are required. I believe that null checks should happen as below.
```
return {
        applicationNumber: eachApplication.applicationNumber,
        applicationNumberId: eachApplication.id,
        studyStartPeriod: eachApplication.offering?.studyStartDate ?? "",
        studyEndPeriod: eachApplication.offering?.studyEndDate ?? "",
        coeStatus: eachApplication.coeStatus,
        firstName: eachApplication.student.user.firstName,
        lastName: eachApplication.student.user.lastName,
      };
```